server/mailet/mock-smtp-server/src/main/java/org/apache/james/mock/smtp/server/model/Condition.java

```diff
@@ -75,6 +75,27 @@ public final int hashCode() {
         }
```

Comment:
What happens if I deserialize a condition: `{"operator": "matchNothing"}`? 

`{"operator": "matchNothing", "matchingValue":"aeiou"}`?

Reply:
they all return a `MatchNothingCondition`?

Reply:
`{"operator": "matchNothing", "matchingValue":"aeiou"}` should be rejected as the matching value would be ignored and makes no sens